source/models/creating-and-deleting-records.md

```diff
@@ -40,12 +40,144 @@ store.find('user', 1).then(function(user) {
 });
 ```
 
-### Deleting Records
+When records are first created they start off in the `new` state. This
+means the record only exists on the client and it has yet to be
+persisted to the backend. To persist the record to the backend you
+need to call `record.save()`.
 
-Deleting records is just as straightforward as creating records. Just call `deleteRecord()`
-on any instance of `DS.Model`. This flags the record as `isDeleted` and thus removes
-it from `all()` queries on the `store`. The deletion can then be persisted using `save()`.
-Alternatively, you can use the `destroyRecord` method to delete and persist at the same time.
+```js
+var post = this.store.createRecord('post', {
+  title: 'Rails is Omakase',
+  body: 'Lorem ipsum'
+});
+
+post.save(); // POST to '/posts'
+```
+
+It is important to remember that whenever you create a new record it
+will be added to Ember Data's store. This means even if you don't end
+up persisting the record to the backend it will still show up in the
+list of records returned by `store.all()`. A common pattern is to use
+the route's `resetController` method to clean up records that may no
+longer be needed in the store when exiting a route.
+
+```app/routes/post/new.js
+export default Ember.Route.extend({
+  resetController: function() {
+    var model = this.controller.get('model');
```

Comment:
resetController gets `controller` as the first arg, seems cleaner than looking it up on this.